lib/puppet/provider/templates/webadministration/_setwebsite.ps1.erb

```diff
@@ -0,0 +1,24 @@
+$resource = @{
+  name            = '<%= "#{resource[:name]}" %>'
+  ensure          = '<%= "#{resource[:ensure]}" %>'
+}
+
+function Try-SetItemProperty
```

Comment:
Have pulled this out now to trysetitemproperty.ps1.erb and pulling it in within the create and update methods.